octobot_trading/exchange_data/ohlcv/channel/ohlcv_updater.py

```diff
@@ -113,12 +115,24 @@ async def _get_init_candles(self, time_frame, pair):
         self.channel.exchange_manager.exchange.uniformize_candles_if_necessary(candles)
         return candles
 
-    async def _initialize_candles(self, time_frame, pair, should_retry) -> (str, common_enums.TimeFrames, list):
+    async def _initialize_candles(self, time_frame, pair, should_retry, wait_for_dependencies) \
+            -> (str, common_enums.TimeFrames, list):
         """
         Manage timeframe OHLCV data refreshing for all pairs
         :return: a tuple with (trading pair, time_frame, fetched candles)
         """
         self._set_initialized(pair, time_frame, False)
+        if wait_for_dependencies:
+            # do not push candles without an initialized price
```

Comment:
Why ?

Reply:
this is because ohlcv are triggering evaluators and trading modes require an initialized price to work. I'm not really happy with this change but I don't now know where to put it exactly. The idea is that we should not start the evaluation chain before the price is initialized. And we don't "just" want to wait for the price because the price might arrive after candles and in this case, that would mean that we would skip the initial evaluation.
Or maybe we add a wait for price init on TA ohlcv consumers ?

Reply:
I understand the goal of this but I really don't think this is the proper place to put it.
Yes I think It would make much more sense in TA ohlcv consumers. 
If we add this line here we cut the ability to receive OHLC to anyone that would need it before price is set